### ISSUE TITLE
JFormFieldFolderList returning pull path as default

### ISSUE THREAD

**Issue by shamsbd71** [joomla/joomla-cms#11308]
I was using t3 framework and for its theme list, its used class from JFormFieldFolderList. but after updating to latest joomla, its started passing full path instead of the folder name.

#### Steps to reproduce the issue
I'm not pretty sure in core joomla where its been used. but as if you try this:
https://docs.joomla.org/Folderlist_form_field_type

you will get folders name with full path.

#### Expected result
We should get only folder name, not full path. and when we will need the full path, we should have options to pass the parameter, and it will inharit the value from extended class.

#### Actual result

showing full path after updating to joomla new version 3.6
![image](https://cloud.githubusercontent.com/assets/1936565/17128888/9b2afdb6-5331-11e6-9c58-ad6e171a5233.png)

#### System information (as much as possible)
PHP 5.6, php 7.0
joomla 3.6

#### Additional comments

if we change this code : https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/form/fields/folderlist.php#L204

instead of this: 
```
$folders = JFolder::folders($path, $this->filter, $this->recursive, true);
```
use 
```
$folders = JFolder::folders($path, $this->filter, $this->recursive, false);
```
will solve the problem. 

thanks

**Comment by laoneo** [comment]
Please test PR #11288 as it solves your issue.

**Comment by brianteeman** [comment]
Closed as a duplicate report - please see #11288
